src/components/ListItem.jsx

```diff
@@ -19,6 +20,30 @@ export function ListItem({ item, listId }) {
 		return false;
 	};
 
+	function determineItemIndicator(item) {
+		const now = new Date();
+		if (
+			item.dateLastPurchased &&
+			getDaysBetweenDates(item.dateLastPurchased.toDate(), now) > 60
+		) {
+			return 'inactive';
+		}
+		const days = getDaysBetweenDates(now, item.dateNextPurchased.toDate());
+		if (days < 0) {
+			return 'overdue';
+		} else if (days <= 7) {
+			return 'soon';
+		} else if (days > 7 && days <= 30) {
+			return 'kind of soon';
+		} else {
+			return 'not soon';
+		}
+	}
+	const indicatorClass = `indicator-${determineItemIndicator(item).replaceAll(
```

Comment:
Just a curiosity/best practice comment, no need to revise. I wonder if it's more React best practice to declare this state to be at the top of the component and within a useState hook. If done this way we might need to use a useEffect hook to trigger setting the state when the component mounts......... really not sure if this is totally off.

Reply:
are you asking about indicatorClass? It isn't state it is just a variable to use to get the css class for the indicator? I believe if I understand your questions correctly?

Reply:
Great question @patty-rose! I think that this is better off as a variable b/c if we were to declare as state and then update that state every time that `item` updates, then we'd be causing an unnecessary re-render. Also, listening for `item` updates via the dependency array gets trigger b/c as an object it would technically retrigger the dependency array on every render unless we did some fancy stuff.

Long story short, I usually try to minimize declaring things as state unless it's an operation that requires a bunch of processing power and want to run that function as little as possible rather than on every re-render. In this case it's great as is. :-)